beacon_chain/utils/hashing/src/merkle_root.rs

```diff
@@ -0,0 +1,31 @@
+use types::{Hash256}
+use hashing::canonical_hash;
+use std::iter::range_step;
+
+fn merkle_root(values: Vec<T>) -> Hash256 {
+    let mut o = vec![0; values.len()];
```

Comment:
these 'zero' items are just place holders to hold our hash results with the merkle root ending up at index 1 - that is, o[1]